scripts/apps/search/directives/RepoDropdown.js

```diff
@@ -2,10 +2,14 @@
 export function RepoDropdown() {
     return {
         scope: {
-            providers: '=',
             activeProvider: '=',
             toggleProvider: '=',
         },
         template: require('../views/repo-dropdown.html'),
+        link: (scope) => {
+            scope.isActive = (provider) => scope.activeProvider ?
+                scope.activeProvider._id === provider._id :
+                !provider._id;
```

Comment:
Negating non-boolean values makes it hard to understand what are you doing. It'd be simpler if you used `provider._id == null` which is what I guess you actually mean by `!provider._id`. I think the same stands for type coercion in general, so you could also improve the ternary by adding a value to compare to `scope.activeProvider != null ?`